### PR TITLE
Translate `State as a Snapshot` page to Indonesia

### LINKED ISSUE #448: State as a Snapshot
Page: https://id.react.dev/learn/state-as-a-snapshot

Assignee: @lionelritchie29

### PR DESCRIPTION
**Pull request by lionelritchie29**
Closes #448 

## Description

Translate the **State as a Snapshot** page.
Page URL: https://id.react.dev/learn/state-as-a-snapshot

### CODE REVIEW

**Review by r17x — APPROVED** [review]

**Review by mhaidarhanif — COMMENTED** [review]
Should be fine but let's wait for the others. @resir014 @mazipan @zainfathoni

**Review by mhaidarhanif — APPROVED** [review]
Should be fine but let's wait for the others. @resir014 @mazipan @zainfathoni

File: src/content/learn/state-as-a-snapshot.md [hunk 1]
```diff
@@ -61,43 +61,43 @@ label, textarea { margin-bottom: 10px; display: block; }
 
 </Sandpack>
 
-Here's what happens when you click the button:
+Inilah yang terjadi ketika Anda menekan tombol "Kirim" pada contoh diatas:
```
Comment (mazipan) [inline src/content/learn/state-as-a-snapshot.md:64]:
```suggestion
Inilah yang terjadi ketika Anda menekan tombol "Kirim" pada contoh di atas:
```

File: src/content/learn/state-as-a-snapshot.md [hunk 2]
```diff
@@ -61,43 +61,43 @@ label, textarea { margin-bottom: 10px; display: block; }
...
-3. React re-renders the component according to the new `isSent` value.
+1. *Event handler* `onSubmit` dijalankan.
+2. `setIsSent(true)` mengubah nilai `isSent` menjadi `true` dan memasukkan antrian *render* baru.
+3. React melakukan *render* ulang pada komponen tersebut sesuai dengan nilai `isSent` yang baru.  
 
-Let's take a closer look at the relationship between state and rendering.
+Mari kita lihat lebih dekat hubungan antara *state* dan *rendering*.
 
-## Rendering takes a snapshot in time {/*rendering-takes-a-snapshot-in-time*/}
+## Rendering mengambil sebuah snapshot pada waktu tersebut {/*rendering-takes-a-snapshot-in-time*/}
```
Comment (mazipan) [inline src/content/learn/state-as-a-snapshot.md:72]: Ini enaknya apa ya? `Snapshot in time` @mhaidarhanif @zainfathoni @r17x @resir014 ?

File: src/content/learn/state-as-a-snapshot.md [hunk 3]
```diff
@@ -139,18 +139,18 @@ Notice that `number` only increments once per click!
...
+    - React mempersiapkan untuk mengubah `number` menjadi `1` pada *render* selanjutnya.
 
-Even though you called `setNumber(number + 1)` three times, in *this render's* event handler `number` is always `0`, so you set the state to `1` three times. This is why, after your event handler finishes, React re-renders the component with `number` equal to `1` rather than `3`.
+Walaupun Anda memanggil `setNumber(number + 1)` sebanyak tiga kali, dalam *event handler* pada *render* ini nilai `number` selalu `0`, sehingga Anda mengubah *state* tersebut menjadi `1` sebanyak tiga kali. Inilah sebabnya, setelah *event handler* Anda selesai dijalankan, React melakukan *render* ulang pada komponen tersebut dengan `number` bernilai `1`, bukan `3`. 
 
-You can also visualize this by mentally substituting state variables with their values in your code. Since the `number` state variable is `0` for *this render*, its event handler looks like this:
+Anda juga dapat memvisualisasikan hal ini pada benak Anda dengan mengganti variabel *state* dengan nilai aselinya pada kode Anda. Karena variabel *state* `number` adalah `0` untuk *render* ini, *event handler*-nya terlihat seperti ini:
```
Comment (mazipan) [inline src/content/learn/state-as-a-snapshot.md:153]:
```suggestion
Anda juga dapat memvisualisasikan hal ini pada benak Anda dengan mengganti variabel *state* dengan nilai aslinya pada kode Anda. Karena variabel *state* `number` adalah `0` untuk *render* ini, *event handler*-nya terlihat seperti ini:
```

File: src/content/learn/state-as-a-snapshot.md (hunk 2 above)
Comment (resir014) [reply src/content/learn/state-as-a-snapshot.md:72]:
```suggestion
## Rendering mengambil sebuah snapshot pada waktu render tersebut {/*rendering-takes-a-snapshot-in-time*/}
```

Based on the translation a few lines after this, we can do it like this.

**Review by resir014 — CHANGES_REQUESTED** [review]
Some changes requested. @lionelritchie29

**Comment by lionelritchie29** [comment]
> Some changes requested. @lionelritchie29

I think it is okay to change it to that. Should I commit the suggestion or it must be done by the maintainer? @resir014

**Comment by mazipan** [comment]
You fix it by yourself, then we will re-review after your fixes.

**Comment by lionelritchie29** [comment]
> You fix it by yourself, then we will re-review after your fixes.

I've updated the content as suggested. 

Thank you.

**Review by resir014 — APPROVED** [review]

**Pull request merged**